xonsh/readline_shell.py

```diff
@@ -15,6 +15,7 @@
 import cmd
 import time
 import select
+import termios
```

Comment:
This isn't sufficiently cross platform.  You'd probably want to move the lazy imports from teepty ([see here](https://github.com/xonsh/xonsh/blob/master/xonsh/teepty.py#L25)) to lazyimps.py, and then import from lazyimps.py in both teepty and this module.